### PR TITLE
[consensus] add unit test for the echo timeout

### PR DESCRIPTION
**Pull request by zekun000**
### Test Plan

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2335)

### CODE REVIEW

File: consensus/src/round_manager_test.rs [hunk 1]
```diff
@@ -1022,3 +1022,48 @@ fn safety_rules_crash() {
...
+        }
+        // timeout 3 nodes
+        for node in &mut nodes[1..] {
+            node.round_manager
+                .process_local_timeout(1)
+                .await
+                .unwrap_err();
+        }
+        let node_0 = &mut nodes[0];
+        // node 1 doesn't timeout and should echo the timeout after 2 timeout message
```
Comment (sitalkedia) [inline consensus/src/round_manager_test.rs:1045]: Why 2 timeout messages? Node 0 should wait for 3 timeout messages for 2f + 1 timeout certficate?

Comment (zekun000) [reply consensus/src/round_manager_test.rs:1045]: this is to test the echo timeout behavior added in https://github.com/aptos-labs/aptos-core/pull/2187

Comment (ikabiljo) [reply consensus/src/round_manager_test.rs:1045]: you mean node 0, not node 1?

**Review by sitalkedia — APPROVED** [review]

File: consensus/src/round_manager_test.rs [hunk 2]
```diff
@@ -1022,3 +1022,48 @@ fn safety_rules_crash() {
...
+        for i in 0..2 {
```
Comment (ikabiljo) [inline consensus/src/round_manager_test.rs:1046]: should we check that it doesn't send the echo timeout again after receiving a 3rd timeout?

**Review by ikabiljo — APPROVED** [review]

**Pull request merged**